.circleci/config.yml

```
@@ -38,91 +39,42 @@ jobs:
     steps:
       - checkout
       - restore_cache:
-          key: go-cache-1-14
+          key: go-cache-1.14
       - run:
           name: Install golangci-lint linter
           command: curl -sfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh| sh -s -- -b $(go env GOPATH)/bin v1.25.0
       - run:
           name: Check formatting
           command: make lint
-  test-1-13:
-    docker:
-      - image: circleci/golang:1.13
-    steps:
-      - checkout
-      - restore_cache:
-          key: go-cache-1-13
-      - credentials
-      - test:
-          go_version: "1.13"
-      - save_cache:
-          key: go-cache-1-13
-          paths:
-            - "~/go/pkg"
-  test-1-14:
-    docker:
-      - image: circleci/golang:1.14
-    steps:
-      - checkout
-      - restore_cache:
-          key: go-cache-1-14
-      - credentials
-      - test:
-          go_version: "1.14"
-      - save_cache:
-          key: go-cache-1-14
-          paths:
-            - "~/go/pkg"
-  test-1-15:
-    docker:
-      - image: circleci/golang:1.15
-    steps:
-      - checkout
-      - restore_cache:
-          key: go-cache-1-15
-      - credentials
-      - test:
-          go_version: "1.15"
-      - save_cache:
-          key: go-cache-1-15
-          paths:
-            - "~/go/pkg"
-  test-1-16:
-    docker:
-      - image: circleci/golang:1.16
-    steps:
-      - checkout
-      - restore_cache:
-          key: go-cache-1-16
-      - credentials
-      - test:
-          go_version: "1.16"
-      - save_cache:
-          key: go-cache-1-16
-          paths:
-            - "~/go/pkg"
-  test-1-17:
+
+  test:
+    parameters:
+        go_version:
+          type: string
     docker:
-      - image: circleci/golang:1.17
+      - image: circleci/golang:<< parameters.go_version >>
```

```suggestion
      - image: cimg/golang:<< parameters.go_version >>
```
Replace the legacy `circleci/golang` with `cimg/golang`.

---

Yep I was going to do it in a follow-up PR, but I can do it now.